files/grest/cron/jobs/populate-next-epoch-nonce.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+DB_NAME=cexplorer
+NWMAGIC=
+EPOCH_LENGTH=
+PROM_URL=
+
+export CARDANO_NODE_SOCKET_PATH=/opt/cardano/cnode/sockets/node0.socket
```

Comment:
Is this the correct way to do it if we need to use `cardano-cli`? Would need to use `sed` here as well similarly to the above constants.

Reply:
For consistency, would be good to have same format (User Variables vs Do Not modify sections), where part below are populated by setup-grest.sh , and -f would cause variables to be overwritten using checkUpdate function from env.

Alternatively, we could populate files when setup-grest.sh is run based on env params

Reply:
We also need to give it the `cardano-cli` path, so I suggest we use `setup-grest.sh` for this as it is only related to a cron job and is easily handled in that script.

```
[[ ${job} == populate-next-epoch-nonce ]] &&
      sed -e "s@NWMAGIC=.*@NWMAGIC=${NWMAGIC}@" -i "${CRON_SCRIPTS_DIR}/${job}.sh" &&
      sed -e "s@EPOCH_LENGTH=.*@EPOCH_LENGTH=${EPOCH_LENGTH}@" -i "${CRON_SCRIPTS_DIR}/${job}.sh" &&
      sed -e "s@PROM_URL=.*@PROM_URL=http://${PROM_HOST}:${PROM_PORT}/metrics@" -i "${CRON_SCRIPTS_DIR}/${job}.sh" &&
      sed for CARDANO_NODE_SOCKET_PATH &&
      sed for CARDANO_CLI_PATH  
```

Reply:
I understand @rdlrt point of user variables for consistency but I dont think these scripts change much from default as variables are properly populated by setup-grest.sh

So I think it's fine to continue as @dostrelith678 suggested.

But we need to add CCLI variable populated with the correct absolute path.
Also, is CARDANO_NODE_SOCKET_PATH path replaced by setup-grest to correct path, if cnode is not used as path?

Reply:
Yes, absolute path for CCLI, I am thinking just `$(which cardano-cli)`, assuming user running `setup-grest.sh` knows about it.

`CARDANO_NODE_SOCKET_PATH` I guess depends only on the vname for cnode... Is there any better way to detect it? I guess we could scan the `cardano-node` process for the socket argument. In that case, this step would fail if `cardano-node` is actually not running for some reason.

Reply:
setup-grest.sh source env file, so CCLI variable and SOCKET variable is already available in setup-grest to be used to replace variable in cron script. I would use this over finding bin from PATH.